ui-frontend/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/cesium/drawing-and-display.tsx

```diff
@@ -61,6 +61,123 @@ export const removeOldDrawing = ({ map, id }: { map: any; id: string }) => {
   relevantPrimitives.length > 0 && map.getMap().scene.requestRender()
 }
 
+const makeOldDrawingNonEditable = ({ map, id }: { map: any; id: string }) => {
+  const relevantPrimitives = map
+    .getMap()
+    .scene.primitives._primitives.filter((prim: any) => {
+      return prim.id === id
+    })
+  relevantPrimitives.forEach((relevantPrimitive: any) => {
+    if (typeof relevantPrimitive.setEditMode === 'function') {
+      relevantPrimitive.setEditMode(false)
+    }
+    if (typeof relevantPrimitive.setEditable === 'function') {
+      relevantPrimitive.setEditable(false)
+    }
+  })
+  relevantPrimitives.length > 0 && map.getMap().scene.requestRender()
+}
+
+const nestedArraysOverlap = (arrayA: any[], arrayB: any[]) => {
+  let result = false
+  return arrayA.some((elemA) => arrayB.some((elemB) => JSON.stringify(elemA) === JSON.stringify(elemB)))
+    result =
+      result ||
+      arrayB.some((elemB) => JSON.stringify(elemA) === JSON.stringify(elemB))
+  })
+  return result
+}
```

Comment:
Oops, I didn't do that suggestion right I guess... I think this would work though:
```suggestion
const nestedArraysOverlap = (arrayA: any[], arrayB: any[]) => {
  return arrayA.some((elemA) => arrayB.some((elemB) => JSON.stringify(elemA) === JSON.stringify(elemB)))
}
```